python/tank/util/pyside6_patcher.py

```diff
@@ -210,27 +226,33 @@ def emit(screen_index):
 
 
         original_QScreen_availableGeometry = QtGui.QScreen.availableGeometry
-        def availableGeometry(self, widget=None):
-            """Patch QScreen to also act as QDesktopWidget."""
-            if widget is None:
+        def availableGeometry(self, arg__1=None):
+            """
+            Patch QScreen to also act as QDesktopWidget.
+
+            :param arg__1 Union(int, QtGui.QWidget, QtCore.QPoint): A widget, screen index or point.
+            """
+            if arg__1 is None:
                 return original_QScreen_availableGeometry(self)
 
-            if isinstance(widget, int):
+            if isinstance(arg__1, int):
                 screens = QtGui.QGuiApplication.screens()
                 try:
-                    screen = screens[widget]
+                    screen = screens[arg__1]
                 except IndexError:
                     return QtCore.QRect()
+            elif isinstance(arg__1, QtCore.QPoint):
+                return original_QScreen_availableGeometry(self)
             else:
-                screen = widget.screen()
+                screen = arg__1.screen()
 
             return screen.availableGeometry()
 
         def screenNumber(self, widget):
             """Provide QDesktopWidget method through QScreen."""
 
-            screen = widget.screen()
             try:
+                screen = widget.screen()
```

Comment:
Curious why we moved this line into the try/except block, are we trying to catch an `IndexError` on the call to `widget.screen()`? Or is there a different exception it could raise?

Reply:
Or you can add an except block for `AttributeError` if the `widget` does not have the attribute `screen`